lib/internal/test_runner/mock/fake_timers.js

```diff
@@ -56,18 +72,27 @@ class FakeTimers {
 
     this.now += time;
     const timers = this.fakeTimers.timers;
+    const alreadyProcessed = new SafeSet();
+    while (true) {
+      const timer = timers.peek();
```

Comment:
Isn't this essentiailly the following?
```js
do 
for(let timer = timers.peek(); timer.runAt <= this.now; timer = timers.peek()) {
  timer.callback(...timer.args);
  timers.shift();
  if (timer.interval) {
    timer.runAt = timer.runAt + timer.interval; (// save the interval duration on the timer when adding a timer)
    timers.insert(timer);
  }
}
```

That would also let you avoid the whole position metaddata saving

Reply:
Nice! worked for setTimeout. Inspecting for interval